libtalek/client_test.go

```diff
@@ -70,4 +76,49 @@ func TestWrite(t *testing.T) {
 	}
 }
 
-//TODO: test reading.
+func TestRead(t *testing.T) {
+	config := ClientConfig{
+		&common.CommonConfig{NumBuckets: 64, BucketDepth: 4, DataSize: 1024, BloomFalsePositive: 0.05, MaxLoadFactor: 0.95, LoadFactorStep: 0.05},
+		time.Second,
+		time.Second,
+		[]*common.TrustDomainConfig{common.NewTrustDomainConfig("TestTrustDomain", "127.0.0.1", true, false)},
+	}
+
+	reads := make(chan *common.EncodedReadArgs, 1)
+	leader := mockLeader{nil, reads}
+
+	c := NewClient("TestClient", config, &leader)
+	if c == nil {
+		t.Fatalf("Error creating client")
+	}
+
+	handle, _ := NewTopic()
+
+	// Recreate the expected buckets to make sure we're seeing
+	// the real write.
+	var seqNoBytes [12]byte
```

Comment:
factored out a nextBucket method to reduce the complexity of re-generating the next buckets